src/main/groovy/com/github/jk1/license/reader/ConfigurationReader.groovy

```diff
@@ -74,7 +74,9 @@ class ConfigurationReader {
                 LOGGER.debug("Collecting dependency ${root.name}")
                 accumulator.add(root)
             }
-            root.children.each { collectDependencies(accumulator, visitedExcludes, it)}
+            if (config.includeTransitiveDeps) {
+                root.children.each { collectDependencies(accumulator, visitedExcludes, it) }
```

Comment:
We really should have some tests for this.
I'm not 100% certain if this works as expected. I guess you do because you tried it out manually, but having a test for this really would help...

Reply:
So true. Besides, I'd rather implement this as a filter for testability's sake.